hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -113,6 +124,16 @@ public void initialize(String inputName, Map<String, String> properties) {
     this.clients = new CachedClientPool(conf, properties);
   }
 
+  @Override
+  public TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
+    return new ViewAwareTableBuilder(identifier, schema);
```

Comment:
Couldn't find a better way to do the validations with other contents if exists or not ?This validation should must happen before the refresh operation.